core/src/main/java/org/elasticsearch/http/HttpServer.java

```diff
@@ -71,15 +65,23 @@ protected void doStart() {
         if (logger.isInfoEnabled()) {
             logger.info("{}", transport.boundAddress());
         }
+        registerHttpAddress();
```

Comment:
I agree with your conclusion but not with your premise. This is a lifecycle method and not a constructor.